src/client/http/connection.rs

```diff
@@ -108,7 +108,6 @@ impl HttpError {
                 } else if e.is_timeout() {
                     kind = HttpErrorKind::Timeout;
```

Comment:
q: would it still make sense to break after a `kind` is found at any level of the error chain rather than continuing and potentially overwriting something higher up the stack with something lower?